File: ProyectoIOS/ProyectoIOS/Profesor.swift
//
//  Profesor.swift
//  ProyectoIOS
//
//  Created by dam on 27/1/17.
//  Copyright © 2017 dam. All rights reserved.
//

import Foundation

class Profesor{
    private var id: Int
    private var nombre: String
    
    init(id:Int , nombre:String) {
        self.id =  id
        self.nombre = nombre
    }
    
    init() {
        self.id =  0
        self.nombre = ""
    }
    
    func getId()->Int{
        return self.id
    }
    
    func setId(id:Int){
        self.id = id
    }
    
    func getNombre()->String{
        return self.nombre
    }
    
    func setNombre(nombre:String){
        self.nombre = nombre
    }
}

File: servidor rest/clases/__doctrine/src/Profesor.php
<?php
use Doctrine\Common\Collections\ArrayCollection;

/**
 * @Entity 
 * @Table(name="profesor")
 */
class Profesor{
    /**
     * @Id
     * @Column(type="integer") @GeneratedValue
     */
    protected $id;
    
    /**
     * @Column(type="string", length=50, unique=false, nullable=false)
     */
    protected $nombre;
    
    /**
     * @OneToMany(targetEntity="Actividad", mappedBy="profesor")
     */
    protected $idp = null;
    
    public function __construct() {
        $this->idp = new ArrayCollection();
    }

    
        public function getId() {
        return $this->id;
    }

    public function getNombre() {
        return $this->nombre;
    }

    public function setId($id) {
        $this->id = $id;
        return $this;
    }

    public function setNombre($nombre) {
        $this->nombre = $nombre;
        return $this;
    }
    
    public function addIdp($idp){
        $this->idp[] = $idp;
    }
    
    function __toString(){
        return 'Profesor: ' . $this->getId() . ' ' . $this->getNombre();
    }
}
